documentation/src/main/asciidoc/topics/proc_creating_protostream_adapter.adoc

```diff
@@ -0,0 +1,22 @@
+[id='creating-protostream-adapter_{context}']
+= Creating ProtoStream adapter classes
+
+ProtoStream provides a `@ProtoAdaptor` annotation that you can use to handle external, third-party Java object classes that you cannot annotation directly.
+
+[TIP]
+====
+Add the `protostream-types` artifact to your project and use the `Adapter` class to get additional type collections.
```

Comment:
No need to explicitly add `protostream-types` dependency as included in `core` and the java client.

Reply:
as said before without that it is not compiling, if this is a mistake we should have a Jira

Reply:
@wfink please raise a Jira or mention it on zulip if this is not working as described.

Reply:
@ryanemerson @oraNod 
as far as I understood the discussion with @anistor it is not wanted to scann the complete classpath for @Proto annotations.
This is the reason why you need to add it if you use fields which need that.

Reply:
@wfink Looking at your `infinispan.playground.encoding` repo, the reason your code doesn't compile without the `protostream-types` dependency, is because you have created the annotated POJOs in a separate domain package that does not include a `infinispan-core` or `infinispan-client-hotrod` dependency. See this [PR](https://github.com/wfink/infinispan.playground.encoding/pull/1) as an example that the module will compile if `protostream-types` is not explicitly provided, but the core or client jars are on the classpath.

Reply:
Let's rephrase here and say:

"{brandname} provides all adapter classes in the `org.infinispan.protostream.types` package, which is available with the `infinispan-core` and `infinispan-client-hotrod` dependencies."

Reply:
@oraNod I'd say "{brandname} provides adapter classes for some common JDK classes in the `protostream-types` artifact, which is included in the infinispan-core and infinispan-client-hotrod dependencies. Please check the [javadoc](link to org.infinispan.protostream.types package javadoc)" for the full list of provided adapters".